Examples/test-suite/javascript/Makefile.in

```diff
@@ -84,7 +84,11 @@ ifeq (node,$(JSENGINE))
 
 	run_testcase = \
 		if [ -f $(srcdir)/$*$(SCRIPTSUFFIX) ]; then \
-			env NODE_PATH=$$PWD:$(srcdir) $(RUNTOOL) $(NODEJS) $(srcdir)/$*$(SCRIPTSUFFIX); \
+			export NODE_PATH=$$PWD:.; cd $(srcdir); \
+			$(RUNTOOL) $(NODEJS) $*$(SCRIPTSUFFIX); \
+			if [ -f $*_workers$(SCRIPTSUFFIX) ]; then \
+				$(RUNTOOL) $(NODEJS) $*_workers$(SCRIPTSUFFIX); \
+			fi \
```

Comment:
I really don't want any complications in the makefile in the generic case and I'd rather keep the environment variable setting pattern we have using env instead of changing to export. I'd prefer any complications went into an individual runme.js file. I think I've missed something completely here, but can you not put Worker testing into another testcase that does not already have a runme.js file?

Reply:
One can enclose it in `()` to prevent environment variable "leakage". Just in case, rationale was that normal make invokes separate shell for each line, so there won't be any actual leakage to next test. Of course one might wonder if there is some make variable that would execute everything in single shell... Either way, should I add `()`? Or should I think of something completely different?

As for putting Worker test into separate testcase. Well, I still need a swig module to `require`, and it's only natural to reuse one that is already compiled. Then one needs a script to pass to `new Worker()`-s as argument, and it's only natural to reuse one that is already available. Of course one can pass a string to `new Worker()`-s and complement it with `{eval: true}`, I've played with this option, but found it somewhat ugly. Simply grabbing an existing script felt more natural and was more elegant.

Note that we can skip this commit and do it in separate PR...

Reply:
> Or should I think of something completely different?

Most notably if approach to check for presence of a "side-kick" `$*_workers$(SCRIPTSUFFIX)` is generally acceptable?

Reply:
> As alternative and more extreme approach we can consider testing **all** `$*$(SCRIPTSUFFIX)`-s with workers. By first executing `$(NODEJS) $*$(SCRIPTSUFFIX)` and then `$(NODEJS)  $(srcdir)/workers.js $*$(SCRIPTSUFFIX)`, where workers.js would instantiate whichever script passed on the command line instead of hard-coding specific one...

For reference, such test with v12 results in few SEGV and unhandled exception in template_static. And is <1% slower. As initially mentioned, the suggested context-aware init is formally incomplete and needs further work. It's just an initial step, but it's expected and was shown to work with purely computational tasks. In other words some failures can be expected in general case. **When** taken into Workers that is. Which was not even a possibility so far.

Bottom line is that "side-kick" approach can be preferred as intermediate step, till remaining general-case problems are resolved.